File: lang/go/test/vac.fun/algorithms/search/LinearSearch/LinearSearchTest.go
package main

import (
    "vac.fun/algorithms/search/LinearSearch"
    "vac.fun/util/SequenceBuilder"
    "vac.fun/util/TestRunner"
)

func testLinearSearch() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackIncreasing(arr)

    if LinearSearch.Find(arr, -1) != size {
        return false
    }

    if LinearSearch.Find(arr, 2147483647) != size {
        return false
    }

    for i, v := range arr {
        if LinearSearch.Find(arr, v) != i {
            return false
        }
    }

    return true
}

func main() {
    TestRunner.ParseTest(testLinearSearch())
}

File: lang/php/test/vac.fun/algorithms/search/LinearSearchTest.php
<?php
    require_once "vac.fun/algorithms/search/LinearSearch.php";
    require_once "vac.fun/util/SequenceBuilder.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\algorithms\search\LinearSearch;
    use vac\fun\util\SequenceBuilder;
    use vac\fun\util\TestRunner;

    function testLinearSearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (LinearSearch\find($arr, -1) != $size) {
            return false;
        }

        if (LinearSearch\find($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (LinearSearch\find($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testLinearSearch());
    }
?>
